event/src/test/java/com/bazaarvoice/emodb/event/db/astyanax/ChannelAllocationStateTest.java

```diff
@@ -0,0 +1,143 @@
+        SlabRef slabRef = mock(SlabRef.class);
+        when(slabRef.addRef()).thenReturn(slabRef);
+        try {
+            channelAllocationState.attachAndAllocate(slabRef,  Iterators.peekingIterator(sizes.iterator()));
+            _log.info("SUCCESS:  allocated " + Constants.MAX_SLAB_SIZE + " events that exactly filled the slab (" + Constants.MAX_SLAB_SIZE_IN_MEGABYTES + " MB)");
+        } catch (Exception e) {
+            assertTrue(false, "ERROR: " + e.getClass().getName() + " thrown when max # (" + Constants.MAX_SLAB_SIZE + ") events that exactly filled slab allocated");
+        }
+    }
+
+    @Test
+    public void allocateWholeSlabExactBytesLessThanMaxEvents() {
+        int allocSize = Constants.MAX_SLAB_SIZE_IN_BYTES/(Constants.MAX_SLAB_SIZE/2);
+        int lastAllocExtraBytes = 0;
+        if (Constants.MAX_SLAB_SIZE_IN_BYTES % (Constants.MAX_SLAB_SIZE/2) != 0) {
+            lastAllocExtraBytes = Constants.MAX_SLAB_SIZE_IN_BYTES % (Constants.MAX_SLAB_SIZE/2);
+        }
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE/2; i++) {
+            if (i == Constants.MAX_SLAB_SIZE/2-1) {
+                sizes.add(allocSize+lastAllocExtraBytes);
+            } else {
+                sizes.add(allocSize);
+            }
+        }
+        ChannelAllocationState channelAllocationState = new ChannelAllocationState();
+        SlabRef slabRef = mock(SlabRef.class);
+        when(slabRef.addRef()).thenReturn(slabRef);
+        try {
+            channelAllocationState.attachAndAllocate(slabRef, Iterators.peekingIterator(sizes.iterator()));
+            _log.info("SUCCESS:  allocated " + Constants.MAX_SLAB_SIZE/2 + " events that exactly filled the slab (" + Constants.MAX_SLAB_SIZE_IN_MEGABYTES + " MB)");
+        } catch (Exception e) {
+            assertTrue(false, "ERROR: " + e.getClass().getName() + " thrown when 1/2 max # (" + Constants.MAX_SLAB_SIZE/2 + ") events that exactly filled slab allocated");
+        }
+    }
+
+    @Test
+    public void allocateLessThanMaxBytesLessThanMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE-1; i++) {
+            sizes.add(64);
+        }
+        ChannelAllocationState channelAllocationState = new ChannelAllocationState();
+        SlabRef slabRef = mock(SlabRef.class);
+        when(slabRef.addRef()).thenReturn(slabRef);
+        try {
+            channelAllocationState.attachAndAllocate(slabRef, Iterators.peekingIterator(sizes.iterator()));
+            _log.info("SUCCESS:  allocated " + (Constants.MAX_SLAB_SIZE-1) + " 64-byte events");
+        } catch (Exception e) {
+            assertTrue(false, "ERROR: " + e.getClass().getName() + " thrown when max # -1 (" + (Constants.MAX_SLAB_SIZE-1) + ") 64-byte events allocated");
+        }
+    }
+
+    @Test
+    public void allocateSlabInMultipleCalls() {
+        List<Integer> sizes = new ArrayList<Integer>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS2293)